### PR TITLE
fix!: 8.3.6087 intermittent storage CT zero phase data and agggerate value drop

### PR DESCRIPTION
**Pull request by catsmanac**
Yet another Issues with newly pushed Enphase firmware, 8.3.6087 this time. #478 reports that for metered envoy running 8.3.6087  with storage CT in split mode, 2 phase configuration, intermittently data of one phase is all zero and the aggregate values drop to the values of the other phase that still has data.

As a result the aggregate values which are the sum of the 2 phases, switch between the sum of both and the value of the single non-zero phase. The Home Assistant energy dashboard reports spikes in the data because of the switch between 2 different values.

This PR changes:

- The meters updater update method to specifically test for this condition (FW version, storage meter, phase mode and count and zero values in one phase, If this is the case, it returns None for the storage ct aggregate and storage CT zero phase so HA data will show as unavailable rather then wrong data.
- The data model to allow for None entries in the meters data to prevent linter issues.
- Add tests for this specific case to test_ct_meters.py

### This a breaking fix, 

The data model now allows for None entries in the CT meters data. Typed consumers that assume EnvoyMeterData will now fail type checks, and untyped consumers can raise AttributeError on None.

```    
ctmeters: dict[str, EnvoyMeterData | None] = field(default_factory=dict)
ctmeters_phases: dict[str, dict[str, EnvoyMeterData | None]] = field(default_factory=dict)
```

This is also done for the to be deprecated storage data record, if these are still used one should switch to the ctmeters data records. Deprecation may happen shortly.

```
ctmeter_consumption_phases: dict[str, EnvoyMeterData | None] | None = None
ctmeter_storage_phases: dict[str, EnvoyMeterData | None] | None = None
```

> **_NOTE:_**  Do not bump this pyenphase version in Home Assistant without changing the sensor.py code.The native_value function of [EnphaseCtEntity ](https://github.com/catsmanac/ha-core-dev/blob/96092a2dbba2996af0813309a37b78c7a5b07a7f/homeassistant/components/enphase_envoy/sensor.py#L1290-L1296 ) or [EnphaseCtPhaseEntity](https://github.com/catsmanac/ha-core-dev/blob/96092a2dbba2996af0813309a37b78c7a5b07a7f/homeassistant/components/enphase_envoy/sensor.py#L1306-L1320) needs to test for meter data being None. These should test for None and if so return None.

## Bug Fixes

* Improved handling of intermittent zero or unavailable production readings.
* Corrected total and net-consumption calculations when production data is temporarily unavailable.
* Improved storage current transformer handling for affected split- and dual-phase configurations.
* Prevented unreliable aggregate and phase-specific storage measurements from being reported as valid.
* Added support for unavailable meter values in aggregate and phase-specific data.

## Documentation

* Documented affected firmware behavior and clarified when meter readings may be unavailable.
* Added migration guidance for handling nullable meter data.

### CODE REVIEW

**Comment by catsmanac** [comment]
@coderabbitai help

**Comment by bdraco** [comment]
@bluetoothbot review

**Review by bluetoothbot — CHANGES_REQUESTED** [review]
> [!WARNING]
> Important issues found.
>
> - Detection predicate is stricter than the anomaly it describes — may silently never fire
> - Breaking nullability change is not reflected in the published docs

File: tests/fixtures/8.3.6087_storage_ct_drops/ivp_meters_readings [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+                "instantaneousDemand": 0.0,
+                "activePower": 0.0,
+                "apparentPower": 0.0,
+                "reactivePower": 0.0,
+                "pwrFactor": 0.0,
+                "voltage": 0.0,
+                "current": 0.0,
+                "freq": 60.0
+            }
+        ]
```
[the remark this answers is not in the document]
Comment (catsmanac) [reply tests/fixtures/8.3.6087_storage_ct_drops/ivp_meters_readings:210]: > Add a captured affected payload

This has no added value as the patched fixture data is used to override the existing mock from the fixture file and effectively replaces it. The next update call will retrieve the current mock, including the patched data.

[the remark this answers is not in the document]
Comment (bluetoothbot) [reply tests/fixtures/8.3.6087_storage_ct_drops/ivp_meters_readings:210]: Agreed, dropping it. Patched fixture overrides the mock; next update() reads patched data. The real 8.3.6087 fixture you added covers the underlying point — anomaly now synthesized on genuine firmware output, not a version-patched 8.2.x baseline. No further ask here.

**Review by bluetoothbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.

**Review by bluetoothbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.

**Review by bluetoothbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.
